openapi_python_client/parser/properties/model_property.py

```diff
@@ -49,16 +50,56 @@ def get_imports(self, *, prefix: str) -> Set[str]:
         return imports
 
 
+def _is_string_enum(prop: Property) -> bool:
+    return isinstance(prop, EnumProperty) and prop.value_type == str
+
+
+def _is_int_enum(prop: Property) -> bool:
+    return isinstance(prop, EnumProperty) and prop.value_type == int
+
+
+def values_are_subset(first: EnumProperty, second: EnumProperty) -> bool:
+    return set(first.values.items()) <= set(second.values.items())
+
+
+def _is_subtype(first: Property, second: Property) -> bool:
+    from . import IntProperty, StringProperty
+
+    return any(
+        [
+            _is_string_enum(first) and isinstance(second, StringProperty),
+            _is_int_enum(first) and isinstance(second, IntProperty),
+            _is_string_enum(first) and _is_string_enum(second)
+            # cast because MyPy fails to deduce type
+            and values_are_subset(cast(EnumProperty, first), cast(EnumProperty, second)),
+            _is_int_enum(first) and _is_int_enum(second)
+            # cast because MyPy fails to deduce type
+            and values_are_subset(cast(EnumProperty, first), cast(EnumProperty, second)),
+        ]
+    )
```

Comment:
Using `any` like this will cause every single case to be evaluated before `any` is, causing extra work. Here I think the best bet is to refactor into an if/return sequence.

Reply:
Good point. Thanks @dbanty for taking care of this.